src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_index.rs

```diff
@@ -70,14 +69,20 @@ pub fn caml_pasta_fp_plonk_index_create(
         }
         Some(cs) => cs,
     };
-    println!("{}:{}", file!(), line!());
 
     // endo
     let (endo_q, _endo_r) = commitment_dlog::srs::endos::<GAffineOther>();
 
+    // Unsafe if we are in a multi-core ocaml
+    {
+        let ptr: &mut commitment_dlog::srs::SRS<GAffine> =
+            unsafe { &mut *(std::sync::Arc::as_ptr(&srs.0) as *mut _) };
+        ptr.add_lagrange_basis(cs.domain.d1);
+    }
```

Comment:
I really think we should use `RwLock` if we're really doing this. What's the argument not to add these lagrange basis during the creation of the SRS? Or even have a struct of them (like domain) instead of having a hashmap?

Reply:
We don't know ahead of time what size circuits we're going to have. This is safe in OCaml given the global OCaml lock -- would you be OK if I make an issue and do this in the future?

Reply:
that makes sense. I think I also meant: if kimchi as a library is used by other rust libraries, they might need an RwLock on the SRS. Another question is: why not add a lagrange basis on-demand? Every time we try to get a lagrange basis that's not in the hashmap, we create it.